pkg/providers/vsphere/reconciler/reconciler_test.go

```diff
@@ -136,9 +137,9 @@ func TestReconcilerFailToSetUpMachineConfigCP(t *testing.T) {
 
 	result, err := tt.reconciler().ValidateMachineConfigs(tt.ctx, logger, tt.buildSpec())
 	g := NewWithT(t)
-	g.Expect(err).To(MatchError(ContainSubstring("validating vCenter setup for VSphereMachineConfig")))
-	g.Expect(*tt.cluster.Status.FailureMessage).To(ContainSubstring("validating vCenter setup for VSphereMachineConfig"))
-	g.Expect(result).To(Equal(controller.Result{}))
+	tt.Expect(err).To(BeNil(), "error should be nil to prevent requeue")
+	tt.Expect(result).To(Equal(controller.Result{Result: &reconcile.Result{}}), "result should stop reconciliation")
+	g.Expect(tt.cluster.Status.FailureMessage).To(HaveValue(ContainSubstring("validating vCenter setup for VSphereMachineConfig")))
```

Comment:
Is using `g` here instead of `tt` intentional?

Reply:
Fixed